presto-main/src/main/java/com/facebook/presto/operator/ArrayPositionLinks.java

```diff
@@ -47,9 +52,25 @@ public int link(int left, int right)
         }
 
         @Override
-        public Factory build()
+        public PositionLinks.Factory build()
         {
-            return filterFunction -> new ArrayPositionLinks(positionLinks);
+            return new PositionLinks.Factory()
+            {
+                @Override
+                public PositionLinks create(Optional<JoinFilterFunction> joinFilterFunction)
+                {
+                    return new ArrayPositionLinks(positionLinks);
+                }
+
+                @Override
+                public HashCode checksum()
+                {
+                    Hasher hasher = Hashing.goodFastHash(32).newHasher();
```

Comment:
Use `XXHash64`.  If the existing API doesn't work, talk to @martint about changes.